qt-models/divetripmodel.cpp

```diff
@@ -409,6 +418,17 @@ QString DiveItem::displayWeightWithUnit() const
 	return weight_string(weight()) + ((get_units()->weight == units::KG) ? tr("kg") : tr("lbs"));
 }
 
+QString DiveItem::displayTags() const
+{
+	struct dive *dive = get_dive_by_uniq_id(diveId);
+	if (!dive->tag_list)
+		return QString();
+
+	char buf[1024];
+	taglist_get_tagstring(dive->tag_list, buf, 1024);
```

Comment:
i think that our `taglist_get_tagstring()` could have been implemented in a better way.
https://github.com/Subsurface-divelog/subsurface/blob/master/core/dive.c#L3023
if the strings don't fit the buffer size it does a `return i;` which isn't the correct way to approach this generic `fit-this-in-a-buffer` problem.

the return is also never used in our code base.
your code is fine given it follows the our existing usage of `taglist_get_tagstring()` but this function needs changes.

i can take a look at that when i get some free time.